docs/toolhive/concepts/vmcp-architecture.mdx

```diff
@@ -0,0 +1,142 @@
+
+When multiple backends have tools with the same name (for example, both GitHub
+and Jira have `create_issue`), vMCP automatically prefixes them:
+`github_create_issue`, `jira_create_issue`. You can also define custom names for
+clarity.
+
+### Multi-step workflows (composition)
+
+Real-world tasks span multiple systems and require manual orchestration. vMCP
+lets you define declarative workflows with parallel execution, conditionals,
+error handling, and human-in-the-loop approval gates.
+
+**Example scenario**: During an incident investigation, you need logs from your
+logging system, metrics from your monitoring platform, traces from your tracing
+service, and infrastructure status from your cloud provider. With vMCP, you
+fetch all of this in parallel (4x faster than sequential), automatically
+aggregate it into a formatted report, and create a Jira ticket with all the
+data. This workflow is reusable for every incident.
+
+**Example scenario**: For a PR deployment, you might want to merge the PR, wait
+for tests, ask a human for approval, deploy only if approved, and notify Slack.
+vMCP handles this entire flow declaratively, with automatic rollback on
+deployment failure.
+
+### Tool customization and overrides
+
+Third-party MCP servers often have generic names, descriptions, and unrestricted
+parameters. vMCP lets you filter, rename, and wrap tools without modifying the
+upstream servers.
+
+**Security policy enforcement**: You can restrict a fetch tool to internal
+domains only (`*.company.com`), validate URLs before calling the backend, and
+provide clear error messages for policy violations.
+
+**Simplified interfaces**: A complex tool like AWS EC2 might have 20+
+parameters, but your frontend team only needs 3. You can create a simplified
+wrapper that uses instance names instead of IDs and pre-fills safe defaults for
+all other parameters.
+
+### Parameter defaults and pre-configuration
+
+Generic servers require repetitive parameter specification. You always use the
+same repo, channel, or database. vMCP lets you create specialized instances with
+pre-configured defaults.
+
+**Example scenario**: All GitHub queries for your team should default to the
+`stacklok/toolhive` repo. Engineers never need to specify the repo parameter,
+and they can't accidentally query the wrong repository.
+
+**Example scenario**: Staging database is the default (safe for development),
+while production queries require an explicit approval gate. Connection details
+are centralized in the vMCP configuration.
+
+### Authentication boundary separation
+
+Different auth is needed for clients versus backends, and managing credentials
+across multiple systems is complex. vMCP implements a two-boundary auth model
+where clients use SSO and vMCP handles per-backend authentication.
+
+**How it works**: Engineers authenticate once with your identity provider (Okta,
+Google, Microsoft Entra ID). vMCP exchanges their token for backend-specific
```

Comment:
I think it would sound better to not mention token exchange outright, but just say that vMCP authenticates to the back end MCP servers.